metadata-integration/java/acryl-spark-lineage/src/main/java/io/openlineage/spark/agent/vendor/delta/lifecycle/plan/DeltaMergeIntoCommandBuilder.java

```diff
@@ -0,0 +1,161 @@
+      if (tableName == null) {
+        log.warn("Could not extract table name from Delta MergeIntoCommand");
+        return Collections.emptyList();
+      }
+
+      log.info("Detected Delta Lake MERGE INTO command for table: {}", tableName);
+
+      // Create a proper output dataset with the table name using our helper
+      return DeltaTableHelper.getDataset(outputDataset(), tableName, null);
+    } catch (Exception e) {
+      log.error("Error processing Delta Lake MergeIntoCommand", e);
+      return Collections.emptyList();
+    }
+  }
+
+  /** Extracts the target table name from a MergeIntoCommand using reflection. */
+  private String extractTargetTableName(LogicalPlan plan) {
+    try {
+      // Try to get the target field from MergeIntoCommand
+      Method getTarget = plan.getClass().getMethod("target");
+      Object target = getTarget.invoke(plan);
+
+      if (target == null) {
+        return null;
+      }
+
+      // Target could be a LogicalRelation or another relation type
+      // We need to recursively check for common table name fields
+      return extractTableNameFromRelation(target);
+    } catch (Exception e) {
+      log.debug("Could not extract target table name from MergeIntoCommand", e);
+      return null;
+    }
+  }
+
+  /**
+   * Recursively tries to extract table name from a relation object. This handles different versions
+   * of Delta Lake and different relation types.
+   */
+  private String extractTableNameFromRelation(Object relation) {
+    if (relation == null) {
+      return null;
+    }
+
+    try {
+      // Try common field names for table identification
+      for (String methodName :
+          new String[] {"tableName", "tableIdentifier", "identifier", "name", "tablePath"}) {
+        try {
+          Method method = relation.getClass().getMethod(methodName);
+          Object result = method.invoke(relation);
+          if (result != null) {
+            return result.toString();
+          }
+        } catch (NoSuchMethodException e) {
+          // Skip if method doesn't exist
+        }
+      }
+
+      // If relation has a 'relation' field, recursively check it
+      try {
+        Method getRelation = relation.getClass().getMethod("relation");
+        Object nestedRelation = getRelation.invoke(relation);
+        return extractTableNameFromRelation(nestedRelation);
+      } catch (NoSuchMethodException e) {
+        // Skip if method doesn't exist
+      }
+
+      // If relation has a 'table' field, recursively check it
+      try {
+        Method getTable = relation.getClass().getMethod("table");
+        Object table = getTable.invoke(relation);
+        return extractTableNameFromRelation(table);
+      } catch (NoSuchMethodException e) {
+        // Skip if method doesn't exist
+      }
+
+      // Try to get table from 'catalogTable' if it exists
+      try {
+        Method getCatalogTable = relation.getClass().getMethod("catalogTable");
```

Comment:
Can't we identify the class, cast, and use the catalog table method?
I'm afraid of how to detect if Openlineage will rename this method to catalogTable2 or something.